packages/@ember/-internals/meta/lib/meta.ts

```diff
@@ -96,8 +97,20 @@ export class Meta {
     return parent;
   }
 
-  isInitialized(obj: object) {
-    return this.proto !== obj;
+  setInitializing() {
+    this._flags |= MetaFlags.INITIALIZING;
+  }
+
+  unsetInitializing() {
+    this._flags ^= MetaFlags.INITIALIZING;
+  }
+
+  isInitialized() {
```

Comment:
lets name the check here isInitializing then it doesn't have to `not` it, and all the places checking it don't have to `not` it back.